### ISSUE TITLE
Why onFieldsChange is triggered twice?

### ISSUE THREAD

**Issue by allocenx** [react-component/form#156]
I mean, really, why? Seems like **onValuesChange** is not triggered twice like **onFieldsChange**. Basically in **onFieldsChange** im dispatching an action to update redux's state in order move further but this triggering twice results in dispatching twice the same action with same value the only difference being that on first trigger the **errors** is undefined than **errors** has that array of objects with errors... . 

My default trigger is **onChange** which of course results in this twice triggering but **onBlur** does not has this affect. As i have understood from a previous issue seems like this is a **expected behavour**? If so, can anyone explain why?

And further more, can kindly anyone help me  _prevent_ this twice triggering?

Thank you in advance.

EDIT: After further tests i think im starting to understand. Seems like setting the **validateTrigger** to **false** which results in me validating the fields manually, triggers that **onFieldsChange** only once. So i think that the first trigger is for value changing than after the errors pops up aswell it has to trigger again. Therefor, thats why first time the **errors** is undefined and right after its not anymore.  And all of that on **onChange** trigger,  seems like **onBlur** is a whole another story.
All of this seems to make sense now. Please tell if im right and if so what will be a **HACK** to prevent dispatching that action twice?

**Comment by benjycui** [comment]
It is by design.

`field` and `value` are different, `field` includes `errors` and validating status, so it will change when errors and validating status change.
